README.md

```diff
@@ -1,133 +1,616 @@
+    // implementation("software.amazon.smithy:smithy-model:$smithyVersion")
+    // implementation("software.amazon.smithy:smithy-linters:$smithyVersion")
+}
+```
 
-    operation Echo {
-        input := {
-            message: String
-        } 
-        output := {
-            message: String
-        }
-    }
-    ```
+After `smithy-build.json` and `build.gradle.kts` have been configured, run `./gradlew clean build`. This will code generate the TypeScript client under the `source` projection, found in the `build/smithyprojections/quickstart-gradle/source/typescript-client-codegen` directory.
+
+Verify the client is able to compile by running the following:
+
+```shell
+cd build/smithyprojections/quickstart-gradle/source/typescript-client-codegen
+# Yarn is used in this example, but equivalent commands using other package managers can be used, e.g. npm and pnpm
+yarn
+yarn build
+```
+
+> Note that running the NPM scripts to verify the generated TypeScript client is NOT part of the code generation process, and needs to be explicitly executed after the client is generated.
 
-6. Run `gradle build` (alternatively, you can use a
-   [Gradle wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html)).
+For another example of a Gradle project using Smithy Typescript, the `smithy-typescript-codegen-test` package can be referenced as it builds different TypeScript artifacts through projections.
 
-7. The generated client can be found in `build/smithyprojections/foo-client/source/typescript-codegen`.
+See [the Smithy documentation](https://smithy.io/2.0/guides/building-models/gradle-plugin.html) for more information on build Smithy projects with Gradle.
 
-See [the Smithy documentation](https://smithy.io/2.0/guides/building-models/gradle-plugin.html)
-for more information on build Smithy projects with Gradle.
+> Note: the Smithy Gradle Plugin is under heavy development and is subject to breaking changes.
 
 ## TypeScript code generation
 
-TODO
+By default, the Smithy TypeScript code generators provide the code generation framework to generate TypeScript artifacts (e.g. types, interfaces, implementations) of specified Smithy models. However there are implementations for code generation and TypeScript that either need to be implemented or consumed from third-party packages:
+
+- Protocols: Protocols define how operation shapes (for clients and servers, these are usually inputs and outputs) are serialized and deserialized on the wire. This behavior can be defined in Smithy through [protocol traits](https://smithy.io/2.0/spec/protocol-traits.html) with corresponding implementations of [the `ProtocolGenerator` interface](smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/ProtocolGenerator.java). For example, [AWS SDK for JavaScript v3](https://github.com/aws/aws-sdk-js-v3), a customer of Smithy TypeScript, implements the [AWS protocols](https://smithy.io/2.0/aws/protocols/index.html).
```

Comment:
I think the wording of this doesnt quite make it clear that customers can use the existing AWS protocols.

Reply:
Added the dependency name `software.amazon.smithy.typescript:smithy-aws-typescript-codegen`.